src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -40,13 +41,12 @@ def create(self, username, password, stay_signed_in=False):
             logger.debug('Login successful')
             return {'username': username}
 
-        cherrypy.response.status = 403
         if config_username is None:
             logger.warning('No Credentials configured. Need to call `ceph dashboard '
                            'set-login-credentials <username> <password>` first.')
         else:
             logger.debug('Login failed')
-        return {'detail': 'Invalid credentials'}
+        raise DashboardException(msg='Invalid credentials', component='auth')
```

Comment:
Can you add `, code='Invalid credentials'` as well, and distinguish the error by using the code?

Reply:
Done